core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/editor/EditorAgent.java

```diff
@@ -63,10 +62,19 @@
     /**
      * Get all opened editors
      *
-     * @return map with all opened editors
+     * @return list with all opened editors
      */
     @NotNull
-    Map<String, EditorPartPresenter> getOpenedEditors();
+    List<EditorPartPresenter> getOpenedEditors();
+
+    /**
+     * Get opened editor by related file path
+     *
+     * @param path path of the file opened in editor
+     * @return opened editor or null if it does not exist
+     */
+    @Nullable
+    EditorPartPresenter getOpenedEditor(String path);
```

Comment:
>Why did you decide to write so heavy class and used it only once ?
I use it in my branch everywhere, where it possible. It provides usefull support for paths. With less code writing. IMO.

For example, if you have smth like `/path/to/file` and you want to get project from this path, you should create bycicles with substrings, but your path may contains leading slash or not, may contains trailing slash or not. With this utility class you can simple do this: `String project = Path.valueOf("path/to/file").segment(1);` and get need value. Or if you want to get parent of the specified path, you just do this: `Path.valueOf("/some/very/huge/path/to/file").parent()` and get `/some/very/huge/path/to`.

Take a look on eclipse sources, how the developsers use it, and you'll understand. It really helpful for maintaining the resources.